### PR TITLE
Remove Python 2 support from some third-party distributions

### PR DESCRIPTION
**Pull request by srittau**
* Remove all Python 2 only third-party distributions.
* Remove Python 2 support from six as the Python 2 stubs diverged from
  the Python 3 stubs.
* Recursively remove Python 2 support from all distributions that depend
  on the distributions above.

Affects:

* boto
* cryptography
* enum34 (removed)
* ipaddress (removed)
* paramiko
* pysftp
* pyvmomi
* six

Part of #7367

### CODE REVIEW

File: tests/check_new_syntax.py [hunk 1]
```diff
@@ -46,11 +46,13 @@ def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef) -> None:
 
     class ObjectClassdefFinder(ast.NodeVisitor):
         def visit_ClassDef(self, node: ast.ClassDef) -> None:
-            if any(isinstance(base, ast.Name) and base.id == "object" for base in node.bases):
-                errors.append(
-                    f"{path}:{node.lineno}: Do not inherit from `object` explicitly, "
-                    f"as all classes implicitly inherit from `object` in Python 3"
-                )
+            # Temporarily disable this check for two ex-Python 2 stubs.
+            if "paramiko" not in str(path) and "cryptography" not in str(path):
```
Comment (srittau) [inline tests/check_new_syntax.py:50]: I'll send a separate PR once this is merged to fix those problems and re-enable this check.

File: tests/check_consistent.py [hunk 2]
```diff
@@ -68,17 +68,9 @@ def check_stubs():
                 else:
                     assert name.isidentifier(), f"Bad file name '{entry}' in stubs"
             else:
-                if entry in ("@python2", "@tests"):
+                if entry in ("@tests",):
```
Comment (AlexWaygood) [inline tests/check_consistent.py:71]: Any reason not to do this?

```suggestion
                if entry == "@tests":
```

**Review by AlexWaygood — APPROVED** [review]

**Review by Akuli — APPROVED** [review]

**Comment by Akuli** [comment]
These stubs are now Python 3 only, and can potentially be cleaned up:
- boto
- cryptography
- paramiko
- pysftp
- pyvmomi
- six

**Comment by srittau** [comment]
`six` already had separate Python 3 stubs, so I think it's already more or less clean.

**Pull request merged**